Sources/Sentry/SentryTracer.m

```diff
@@ -101,9 +104,117 @@ @implementation SentryTracer {
 #endif // SENTRY_HAS_UIKIT
 }
 
+#if SENTRY_TARGET_PROFILING_SUPPORTED
+static NSObject *appLaunchTraceLock;
+static SentryId *_Nullable appLaunchTraceId;
+BOOL isTracingAppLaunch;
+static uint64_t appLaunchSystemTime;
+SentryTracesSamplerDecision *appLaunchTraceSamplerDecision;
+
+/**
+ * Decision tree:
+ * ```
+ * ─ is there a configuration file?
+ *   ├── yes: SentryTracesSampler sampling decision?
+ *   │   ├── yes: SentryProfilesSampler sampling decision?
+ *   │   │   ├── yes: ✅ trace app launch
+ *   │   │   └── no: ❌ do not trace app launch
+ *   │   └── no: ❌ do not trace app launch
+ *   └── no: ❌ do not trace app launch
+ * ```
+ */
++ (BOOL)shouldTraceAppLaunch
+{
+    [SentryFileManager load];
+
+    NSString *configFilePath = [SentryFileManager sentryLaunchConfigPath];
+
+    if (![[NSFileManager defaultManager] fileExistsAtPath:configFilePath]) {
+        SENTRY_LOG_DEBUG(@"No launch profile config file.");
+        return NO;
+    }
+
+    // get app launch configuration options
+    NSError *error;
+    NSData *configData = [NSData dataWithContentsOfFile:configFilePath options:0 error:&error];
+    if (error != nil) {
+        // we can't read the config file? bail out
+        SENTRY_LOG_DEBUG(@"Couldn't read launch profile config file: %@.", error);
+        return NO;
+    }
+
+    if (configData == nil) {
+        SENTRY_LOG_DEBUG(@"The profile config file was empty.");
+        return NO;
+    }
+
+    error = nil;
+    // a map of strings to numeric values for sample rates or boolean values for
+    // enabling/disabling launch tracing
+    NSDictionary<NSString *, NSNumber *> *launchConfig =
+        [NSJSONSerialization JSONObjectWithData:configData options:0 error:&error];
+    if (error != nil) {
+        SENTRY_LOG_DEBUG(@"Couldn't deserialize launch profile config file: %@", error);
+        return NO;
+    }
+
+    double tracesSampleRate = [launchConfig[@"tracesSampleRate"] doubleValue];
+    double profilesSampleRate = [launchConfig[@"profilesSampleRate"] doubleValue];
+    if (tracesSampleRate == 0 || profilesSampleRate == 0) {
+        SENTRY_LOG_DEBUG(@"Sampling out this launch trace due to sample rate of 0.");
+        return NO;
+    }
+
+    [SentryTracesSampler load];
+    [SentryProfilesSampler load];
+    [SentryRandom load];
+
+    SentryRandom *random = [[SentryRandom alloc] init];
+    appLaunchTraceSamplerDecision = [SentryTracesSampler calcSample:tracesSampleRate random:random];
+    if (appLaunchTraceSamplerDecision.decision != kSentrySampleDecisionYes) {
+        SENTRY_LOG_DEBUG(@"Sampling out this launch trace due to sample decision NO.");
+        return NO;
+    }
+
+    return [[SentryProfilesSampler calcSample:profilesSampleRate random:random] decision]
+        == kSentrySampleDecisionYes;
+}
+
++ (void)startLaunchProfile
```

Comment:
`h`: I'm a bit confused. Nothing here starts the profiler. I think some changes of https://github.com/getsentry/sentry-cocoa/pull/3558 should be here to start the profiler.

Reply:
Good catch, this was a rebase issue from splitting the changes into this and the stacked PR containing tests. I moved it from that branch to this one in https://github.com/getsentry/sentry-cocoa/pull/3529/commits/857facc54e9db9446565a27455593af7861ff862